packages/strapi-plugin-users-permissions/controllers/User.js

```diff
@@ -73,6 +76,10 @@ module.exports = {
         delete ctx.request.body.password;
       }
 
+      if (_.get(ctx.request, 'body.role') && _.get(ctx.request, 'body.role').toString() === '0' && (!_.get(ctx.state, 'user.role') || (_.get(ctx.state, 'user.role') && _.get(ctx.state, 'user.role').toSting() !== '0'))) {
```

Comment:
To make this condition more readable, you can play with the default value provided by the `_.get()` method and there is also a typo on the latest `.toString()` attempt (the `r` was missing).

```js
if (_.get(ctx.request, 'body.role', '').toString() === '0' &&  (!_.get(ctx.state, 'user.role') || (_.get(ctx.state, 'user.role', '').toString() !== '0'))) {
  delete ctx.request.body.role;
}
```